block/manager.go

```diff
@@ -150,18 +175,23 @@ func NewManager(
 		retriever:   dalc.(da.BlockRetriever), // TODO(tzdybal): do it in more gentle way (after MVP)
 		daHeight:    s.DAHeight,
 		// channels are buffered to avoid blocking on input/output operations, buffer sizes are arbitrary
-		HeaderCh:          make(chan *types.SignedHeader, 100),
-		BlockCh:           make(chan *types.Block, 100),
-		blockInCh:         make(chan newBlockEvent, 100),
+		HeaderCh:          make(chan *types.SignedHeader, channelLength),
+		BlockCh:           make(chan *types.Block, channelLength),
+		blockInCh:         make(chan newBlockEvent, channelLength),
+		blockStoreMtx:     new(sync.Mutex),
+		blockStore:        blockStore,
```

Comment:
Follow Up: The block store should be threadsafe on its own.  Can we look into where we are relying on the mutex? 

This mutex might be left over from when we were setting it?

Reply:
The `blockStoreMtx` is used for the `blockStoreCond` for signaling the block store retrieve loop every `blockTime`.

Reply:
Related https://github.com/rollkit/rollkit/issues/1132